dagutils/diff.go

```diff
@@ -112,8 +116,8 @@ func Diff(ctx context.Context, ds ipld.DAGService, a, b ipld.Node) ([]*Change, e
 	}
 
```

Comment:
Not that much (I think), we could add a check here to verify that, if they have links, i.e., they are not raw nodes because they passed the first check (hence we would still be supporting them), they must be proto-nodes; if not, you could also return a `Mod` as before. Actually you could even merge this proto-node check in the previous `if`, WDYT?